### PR TITLE
Fix case where "az aks" errors on empty kubeconfig

### PR DESCRIPTION
**Pull request by mboersma**
**Description**

The `az aks get-credentials` command quits with an error if some default entries aren't present in the user's kubeconfig file. This works around that, since `az capi` shells out to that command.

**History Notes**

---

This checklist is used to make sure that common guidelines for an Azure CLI pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).
- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).
- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/capi/azext_capi/tests/latest/test_helpers.py [hunk 1]
```diff
@@ -365,11 +367,11 @@ class ManagementClusterComponentsMissingMatchExpressionTest(unittest.TestCase):
         "No installation found"
     ]
 
-    def test_valid_matches(self):
+    def test_valid_mgmt_matches(self):
```
Comment (mboersma) [inline src/capi/azext_capi/tests/latest/test_helpers.py:370]: I renamed these to be unique, because there are already test functions named `test_valid_matches` and `test_invalid_matches`. This isn't generally a problem, since they are encapsulated in different test classes.

However, the `azdev test` tool when run with `--discover` complains about the colliding names. This is because its shorthand to run individual tests is simply to specify the function name:

```
% azdev test test_valid_mgmt_matches test_invalid_mgmt_matches
```

I think this is weird but I've never found another syntax to run individual test cases with `azdev test`, so I can go with it.

File: src/capi/azext_capi/custom.py [hunk 2]
```diff
@@ -21,27 +21,26 @@
...
-from .helpers.logger import logger
-from .helpers.spinner import Spinner
-from .helpers.run_command import run_shell_command, try_command_with_spinner
 from .helpers.binary import check_clusterctl, check_kubectl, check_kind
-from .helpers.prompt import get_cluster_name_by_user_prompt, get_user_prompt_or_default
-from .helpers.generic import match_output, is_clusterctl_compatible
-from .helpers.os import set_environment_variables, write_to_file
-from .helpers.network import urlretrieve
 from .helpers.constants import MANAGEMENT_RG_NAME
+from .helpers.generic import has_kind_prefix, match_output, is_clusterctl_compatible
```
Comment (mboersma) [inline src/capi/azext_capi/custom.py:37]: Apologies for the unrelated change; I consolidated and alpha-sorted the imports.

**Review by willie-yao — APPROVED** [review]
LGTM!
